src/components/ChattyLLM/Message.vue

```diff
@@ -97,14 +103,41 @@ export default {
 			displayName: getCurrentUser()?.displayName ?? getCurrentUser()?.uid ?? t('assistant', 'You'),
 			userId: getCurrentUser()?.uid ?? t('assistant', 'You'),
 			showMessageActions: false,
+			// we could initialize this with undefined but then NcRichText will try to find links
+			// and resolve them before we had a chance to do so, resulting in unnecessary requests to references/resolve
+			// with [] we inhibit the extract+resolve mechanism on NcRichText
+			// TODO This can be removed (and all the custom extract+resolve logic) when fixed in NcRichText
+			references: [],
 		}
 	},
 
+	mounted() {
+		this.fetch()
+	},
+
 	methods: {
 		copyMessage(message) {
 			navigator.clipboard.writeText(message)
 			showSuccess(t('assistant', 'Message copied to clipboard'))
 		},
+		fetch() {
+			const urlMatch = (new RegExp(PLAIN_URL_PATTERN).exec(this.message.content.trim()))
+			const mdMatch = (new RegExp(MARKDOWN_LINK_PATTERN).exec(this.message.content.trim()))
+			const firstMatch = urlMatch
+				? urlMatch[1].replaceAll(/[).,:;]+$/g, '')
```

Comment:
```suggestion
				? urlMatch[1].replaceAll(/[).,:;!?]+$/g, '')
```

I would add ! and ? for URLs and sentence ends

Reply:
Done